### PR TITLE
Feat(cluster-api): add watch filter arg

### LINKED ISSUE #330: [Feature]: Support running multiple instances of the same provider
### Explain problem to solve

We need the feature suggested by the cluster-api doc here: https://cluster-api.sigs.k8s.io/developer/architecture/controllers/support-multiple-instances

### Describe the solution you would like

implement the 2 flags `--namespace`and `--watch-filter`

### Additional context

Example implementation on aws provider here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controllers/awscluster_controller.go#L385C3-L385C18

### Environment

```markdown
- Kubernetes version: (use `kubectl version`): -
- OS (e.g. from `/etc/os-release`):
- Kernel (e.g. `uname -a`):
- cluster-api-provider-outscale version:
- cluster-api version:
- Install tools:
- Kubernetes Distribution:
- Kubernetes Diestribution version:
```

### PR DESCRIPTION
**Pull request by sebglon**
**What type of PR is this?**

/kind feature           Add support of watch-filter cluster-api arg

**What this PR does / why we need it**:
Implement the multi controller instance as describe here by cluster-api

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #330 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._
Change contrller image;

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [x] adds unit tests

### CODE REVIEW

File: example/cluster-machine-template-simple.yaml [hunk 1]
```diff
@@ -263,20 +268,22 @@ kind: OscMachineTemplate
...
   template:
     spec:
       node:
         clusterName: cluster-api
         image:
-          name: ubuntu-2004-2004-kubernetes-v1.22.11-2022-08-22
+          name: ubuntu-2004-2004-kubernetes-v1.27.7-2023-10-18
         keypair:
-          name: cluster-api
+          name: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-simple.yaml:281]: hello, I think kaypaire_make, would be a better name here

File: example/cluster-machine-template-simple.yaml [hunk 2]
```diff
@@ -263,20 +268,22 @@ kind: OscMachineTemplate
...
           deleteKeypair: false
         vm:
           clusterName: cluster-api
           name: cluster-api-vm-kw
-          keypairName: cluster-api
+          keypairName: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-simple.yaml:286]: same here

File: example/cluster-machine-template-simple.yaml [hunk 3]
```diff
@@ -294,20 +301,22 @@ kind: OscMachineTemplate
...
   template:
     spec:
       node:
         clusterName: cluster-api
         image:
-         name: ubuntu-2004-2004-kubernetes-v1.22.11-2022-08-22
+         name: ubuntu-2004-2004-kubernetes-v1.27.7-2023-10-18
         keypair:
-          name: cluster-api
+          name: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-simple.yaml:314]: Idem

File: example/cluster-machine-template-simple.yaml [hunk 4]
```diff
@@ -294,20 +301,22 @@ kind: OscMachineTemplate
...
           deleteKeypair: false
         vm:
           clusterName: cluster-api
           name: cluster-api-vm-kcp
-          keypairName: cluster-api
+          keypairName: sebastien
```
Comment (outscale-hmi) [inline example/cluster-machine-template-simple.yaml:319]: idem

**Review by outscale-hmi — COMMENTED** [review]
Thanks for the PR,

**Review by outscale-hmi — APPROVED** [review]
I runned the failed test on local and SGTM,
Thanks for the contribution

**Comment by outscale-hmi** [comment]
@sebglon  Hello, I fixed the boileplate test that failed in your previous commit, can you please rebase and push again and I ll merge as soon as possible
I already tested your changes in local and LGTM

**Comment by sebglon** [comment]
@outscale-hmi rebase is done.

**Comment by outscale-hmi** [comment]
> @outscale-hmi rebase is done.

Thanks for the reactivity, tests are good but still have conflicts to be able to merge

**Comment by sebglon** [comment]
@outscale-hmi  rebase done

**Pull request merged**